lxd/db/cluster/nodes.mapper.go

```diff
@@ -0,0 +1,59 @@
+//go:build linux && cgo && !agent
+
+package cluster
+
+// The code below was generated by lxd-generate - DO NOT EDIT!
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net/http"
+
+	"github.com/lxc/lxd/shared/api"
+)
+
+var _ = api.ServerEnvironment{}
+
+var nodeID = RegisterStmt(`
+SELECT nodes.id FROM nodes
+  WHERE nodes.name = ?
+`)
+
+// GetNodeID return the ID of the node with the given key.
+// generator: node ID
+func GetNodeID(ctx context.Context, tx *sql.Tx, name string) (int64, error) {
+	stmt, err := Stmt(tx, nodeID)
+	if err != nil {
+		return -1, fmt.Errorf("Failed to get \"nodeID\" prepared statement: %w", err)
+	}
+
+	rows, err := stmt.Query(name)
```

Comment:
Why dont we use https://pkg.go.dev/database/sql#Tx.QueryRowContext here? For two reason, firstly we would be passing the context into the query function itself, secondly this function will return at most 1 rows or error if now rows found. The `ErrNoRows` error can be converted into the `api.StatusErrorf(http.StatusNotFound, "Node not found")` error using `errors.Is(err, sql.ErrNoRows)`.

This got me thinking too, do we use https://pkg.go.dev/database/sql#Tx.QueryContext in the DB generator when we do expect multiple rows?